dali/test/python/test_operator_reader_shuffling.py

```diff
@@ -117,13 +168,17 @@ def gather_ids(pipes, epochs_run=0, batch_size=1, num_gpus_arg=None, gpus_arg=No
         iterate_over = gpus_arg
     img_ids_list = [[] for _ in pipes]
 
-    # each GPU needs to iterate from `shard_id * data_size / num_gpus` samples to `(shard_id + 1)* data_size / num_gpus`
+    # each GPU needs to iterate from `shard_id * data_size / num_gpus` samples
+    # to `(shard_id + 1)* data_size / num_gpus`
     # after each epoch each GPU moves to the next shard
-    # epochs_run takes into account that after epoch readers advances to the next shard, if shuffle_after_epoch or stick_to_shard
-    # if doesn't matter and could/should be 0
-    # it is relevant only pad_last_batch == False, otherwise each shard has the same size thanks to padding
+    # epochs_run takes into account that after epoch readers advances to the
+    # next shard, if shuffle_after_epoch or stick_to_shard if doesn't matter
+    # and could/should be 0
+    # it is relevant only if pad_last_batch == False, otherwise each shard has
+    # the same size thanks to padding
```

Comment:
My understanding from the comment and the code. Feel free to copy it or reword it as you see fit:
- there's a variable epochs_run that tells you the epoch count.
- Epochs run is used to know in which shard we currently are. This is because we move to the next shard at the end of the epoch. Here's an example:
3 shards:
Epoch 0: [shard0, shard1, shard2]  # meaning pipelines are reading those shards
Epoch 1: [shard1, shard2, shard0]
Epoch 2: [shard2, shard0, shard1]  # ... and so on
- Such logic does not make sense when shuffle_after_epoch=True or stick_to_shard=True, therefore we just set epochs_run=0 so that it doesn't influence in the shard_size calculation
- Also, when pad_last_batch=False, all shards at the same size, so this logic also becomes irrelevant.